File: core/src/main/java/net/yan100/compose/core/properties/SnowflakeProperties.java
package net.yan100.compose.core.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * @author TrueNine
 * @since 2023-04-01
 */
@Data
@ConfigurationProperties(prefix = "compose.core.snowflake")
public class SnowflakeProperties {
    private Long workId = 1L;
    private Long dataCenterId = 2L;
    private Long sequence = 3L;
    private Long startTimeStamp = 100000L;
}

File: core/src/main/kotlin/net/yan100/compose/core/properties/SnowflakeProperties.kt
package net.yan100.compose.core.properties

import org.springframework.boot.context.properties.ConfigurationProperties

private const val PREFIX = "compose.core"

/**
 * # snowflake id 生成器自动配置
 *
 * @author TrueNine
 * @since 2023-04-01
 */
@ConfigurationProperties(prefix = "$PREFIX.snowflake")
data class SnowflakeProperties(
  var workId: Long = 1L,
  var dataCenterId: Long = 2L,
  var sequence: Long = 3L,
  var startTimeStamp: Long = 100000L,
)
